### PR TITLE
Factory for STSAssumeRoleCredentialsRequestInterceptor to inject implementations extending the interceptor

### LINKED ISSUE #15177: Allow for injection of OAuth2RequestInterceptor and STSAssumeRoleCredentialsRequestInterceptor
**Describe the solution you'd like**
Factory for `OAuth2RequestInterceptor` and `STSAssumeRoleCredentialsRequestInterceptor` to inject implementations extending those two interceptors.

### PR DESCRIPTION
**Pull request by chenkins**
Fixes #15177.

### CODE REVIEW

File: s3/src/main/java/ch/cyberduck/core/sts/STSAssumeRoleAuthorizationService.java [hunk 1]
```diff
@@ -103,11 +104,26 @@ public TemporaryAccessTokens authorize(final String sAMLAssertion) throws Backgr
...
-            log.debug(String.format("Assume role with OIDC Id token for %s", bookmark));
-        }
-        request.setWebIdentityToken(oauth.getIdToken());
+
+
         final HostPreferences preferences = new HostPreferences(bookmark);
+        final String webIdentityToken;
+        // Use OIDC ID token by default. However, OAuth 2.0 access token may also be used:
+        //  see https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html
+        if(StringUtils.isBlank(preferences.getProperty("s3.assumerole.tokentype")) || preferences.getProperty("s3.assumerole.tokentype").equals("id_token")) {
```
Comment (dkocher) [inline s3/src/main/java/ch/cyberduck/core/sts/STSAssumeRoleAuthorizationService.java:113]: Provide default in `default.properties` and omit check for property definition.

Comment (dkocher) [reply s3/src/main/java/ch/cyberduck/core/sts/STSAssumeRoleAuthorizationService.java:113]: Can this be moved into a single pull request.

**Review by dkocher — CHANGES_REQUESTED** [review]

File: s3/src/main/java/ch/cyberduck/core/sts/STSAssumeRoleAuthorizationService.java
Comment (ylangisc) [inline s3/src/main/java/ch/cyberduck/core/sts/STSAssumeRoleAuthorizationService.java:1]: Formatting.

**Comment by dkocher** [comment]
What features are missing in the implementations that require the ability to subclass or inject different implementations?

File: s3/src/main/java/ch/cyberduck/core/sts/STSAssumeRoleAuthorizationService.java
Comment (chenkins) [reply s3/src/main/java/ch/cyberduck/core/sts/STSAssumeRoleAuthorizationService.java:1]: done.

**Comment by dkocher** [comment]
I propose #15227 instead.

**Pull request closed without merging**